<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Welcome to Scipion Chem’s documentation! &mdash; Scipion Chem 1.0.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=19f00094" />
      <link rel="stylesheet" type="text/css" href="_static/custom.css?v=f327017f" />

  
    <link rel="shortcut icon" href="_static/favicon.ico"/>
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js?v=5d32c60e"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="_static/documentation_options.js?v=8d563738"></script>
        <script src="_static/doctools.js?v=9a2dae69"></script>
        <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="scipion-chem" href="plugins/pwchem/index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="#" class="icon icon-home">
            Scipion Chem
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Plugins</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="plugins/pwchem/index.html">scipion-chem</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/amber/index.html">scipion-chem-amber</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/autodock/index.html">scipion-chem-autodock</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/blast/index.html">scipion-chem-blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/durrantlab/index.html">scipion-chem-durrantlab</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/fpocket/index.html">scipion-chem-fpocket</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/gromacs/index.html">scipion-chem-gromacs</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/lephar/index.html">scipion-chem-lephar</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/modeller/index.html">scipion-chem-modeller</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/openmm/index.html">scipion-chem-openmm</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/p2rank/index.html">scipion-chem-p2rank</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/rosetta/index.html">scipion-chem-rosetta</a></li>
<li class="toctree-l1"><a class="reference internal" href="plugins/schrodinger/index.html">scipion-chem-schrodinger</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Scipion Chem enviroment</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="objects/index.html">Objects</a></li>
<li class="toctree-l1"><a class="reference internal" href="tools/automatic-tests/index.html">Automatic testing tool</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Publications</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="publications/scipion-chem_vds/index.html">Scipion-chem VDS Data</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Others</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="contact.html">Contact us</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">Scipion Chem</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content style-external-links">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Welcome to Scipion Chem’s documentation!</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <figure class="align-default" id="home-page">
<img alt="Scipion Chem logo" src="_images/logo.png" />
</figure>
<section id="welcome-to-scipion-chem-s-documentation">
<h1>Welcome to Scipion Chem’s documentation!<a class="headerlink" href="#welcome-to-scipion-chem-s-documentation" title="Link to this heading"></a></h1>
<p>In the following documents, you will find all the necessary information for installing and using scipion-chem and all
the related plugins (called scipion-chem-*). These plugins integrate mainly software for Virtual Drug Screening (VDS).
You can find the complete list in <a class="reference external" href="https://github.com/scipion-chem">https://github.com/scipion-chem</a></p>
<p>Scipion Chem is itself built as a plugin of <a class="reference external" href="https://scipion-em.github.io/docs/release-3.0.0/index.html">Scipion</a>,
a workflow engine mainly used for Cryo-Electron Microscopy.
Therefore, most of its documentation also applies to Scipion-chem and you will need to install
<a class="reference external" href="https://scipion-em.github.io/docs/release-3.0.0/docs/scipion-modes/how-to-install.html">Scipion3</a> first
in order to use Scipion-chem.</p>
</section>
<section id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Link to this heading"></a></h1>
<p>Once Scipion is installed in your computer, you can install Scipion-chem as its plugin. Scipion-chem is the core for the
rest of scipion-chem-* plugins, so you would need to install it first. To do so, you can check the instructions in the
<a class="reference external" href="https://github.com/scipion-chem/scipion-chem/blob/master/README.rst">scipion-chem README</a></p>
<p>Then, you are free to install any subsequent plugin you are interested in. The procedure is very similar, you will find
README files in each of the plugin repositories in case of doubt, appart from a more exhaustive documentation within this pages.</p>
</section>
<section id="scipion-chem-overview">
<h1>Scipion Chem overview<a class="headerlink" href="#scipion-chem-overview" title="Link to this heading"></a></h1>
<p>In this section, each of the Scipion Chem plugins will be briefly presented, grouped by their main functionalities.
We have mainly followed the workflow shown in the figure below, even though some included tools fall outside these
groups. Further documentation will be written for each of them, together with case studies and workflow examples.</p>
<figure class="align-default">
<img alt="vds workflow" src="_images/vds-workflow.gif" />
</figure>
<section id="pwchem-index">
<h2>1) <a class="reference internal" href="plugins/pwchem/index.html#pwchem-index"><span class="std std-ref">scipion-chem</span></a><a class="headerlink" href="#pwchem-index" title="Link to this heading"></a></h2>
<p>As stated before, scipion-chem is the core of the Scipion Chem plugins and it includes several tools which
are mainly oriented to virtual Drug Screening.</p>
<p>scipion-chem automatically installs several util third-party software for the management and visualization of the
tasks in a typical bioinformatics and VDS workflow. These include:</p>
<blockquote>
<div><ul class="simple">
<li><p><a class="reference external" href="https://github.com/openbabel/openbabel">OpenBabel</a> and <a class="reference external" href="https://github.com/rdkit/rdkit">RDKit</a>: the main small molecule handlers and converters.</p></li>
<li><p><a class="reference external" href="https://ccsb.scripps.edu/mgltools/">MGLTools</a>: utadditional utils for small molecules, docking, … (includes <a class="reference external" href="https://autodocksuite.scripps.edu/adt/">AutoDockTools</a>).</p></li>
<li><p><a class="reference external" href="https://jchempaint.github.io/">JChemPaint</a>: Java program to manually draw small molecules.</p></li>
<li><p><a class="reference external" href="https://pymol.org/2/">Pymol</a>: main viewer of Scipion Chem for small molecules and structures.</p></li>
<li><p><a class="reference external" href="https://www.ks.uiuc.edu/Research/vmd/">VMD</a>: secondary viewer of Scipion-Chem for structures and Molecular Dynamics.</p></li>
<li><p><a class="reference external" href="https://github.com/AliView/AliView">AliView</a>: main viewer for sequences.</p></li>
<li><p><a class="reference external" href="https://github.com/pharmai/plip">PLIP</a>: specialized viewer for docking interactions in <a class="reference external" href="https://pymol.org/2/">Pymol</a>.</p></li>
</ul>
</div></blockquote>
<p>Last but not least, Scipion-chem includes 3 groups of protocols: General, Sequences and VirtualDrugScreening. As their
names suggest, they contain Scipion protocols related to general or management tasks, protocols for the management of
sequences and finally those mainly used in a VDS workflow, including both structure and ligand based approaches.
Among them, we include consensus protocols for structural regions of interest (ROIs), which are usually predicted
protein pockets, and for docking results.
Further information about all these protocols can be found in the subsequent documentation.</p>
</section>
<section id="target-and-ligand-preparation">
<h2>2) Target and ligand preparation<a class="headerlink" href="#target-and-ligand-preparation" title="Link to this heading"></a></h2>
<p>The protocols for the protein (target) and the small molecule (ligand) preparation are actually included in other
plugins and do not hold any plugin by themselves. As you will be able to read below, protocols for receptor preparation
can be performed with autodock, rosetta or lephar plugins; while protocols for ligand preparation can be performed using
autodock, openbabel and rdkit (these last two included in Scipion-chem).</p>
</section>
<section id="structural-regions-of-interest-rois">
<h2>3) Structural Regions Of Interest (ROIs)<a class="headerlink" href="#structural-regions-of-interest-rois" title="Link to this heading"></a></h2>
<p>Usually, the VDS workflow from the target starts by defining onto the structure those regions where we are interested
in exploring the druggability. This is, defining those places where we would like to perform the docking.</p>
<p>We can address this task from multiple angles and Scipion-chem tries to offer a flexible environment to do so:</p>
<ul>
<li><div class="line-block">
<div class="line"><strong>Manually define structural ROIs</strong>: If we have some prior knowledge about the receptor, we can just define these structural ROIs from the desired residues, coordinates or even previous ligands.</div>
<div class="line">The user would be able to do so using the “Define Structural ROI” protocol in Scipion-chem.</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Map structural from sequence ROIs</strong>: The user would also be able to map some ROI from the protein sequence to its structure using the “Map sequence ROIs” protocol in Scipion-chem.</div>
<div class="line">For example, one might be interested in mapping conserved regions of the sequence onto the structure</div>
</div>
</li>
<li><div class="line-block">
<div class="line"><strong>Predict binding pocket sites</strong>: Typically, ligand bind to protein pockets or cavities.</div>
<div class="line">We included several plugins including programs which predict the most promising pockets in the protein structure, based on variable factors. These are shown below.</div>
</div>
</li>
</ul>
<section id="scipion-chem-fpocket">
<h3>3.1) <strong>scipion-chem-fpocket</strong><a class="headerlink" href="#scipion-chem-fpocket" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://github.com/Discngine/fpocket">FPocket</a> is one of the most widely known programs for protein pocket detection.
In this plugin we include the main tool for pocket prediction, FPocket, together with the tool for predicting pockets
over a Molecular Dynamics simulation, MDPocket (in devel).</p>
</section>
<section id="scipion-chem-p2rank">
<h3>3.2) <strong>scipion-chem-p2rank</strong><a class="headerlink" href="#scipion-chem-p2rank" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://github.com/rdk/p2rank">P2Rank</a> is a ligand binding site predictor based on machine learning. In this plugin
we include its main functionality for protein pocket prediction.</p>
</section>
</section>
<section id="ligand-based-approaches">
<h2>4) Ligand based approaches<a class="headerlink" href="#ligand-based-approaches" title="Link to this heading"></a></h2>
<p>This section is mainly included in the Scipion-chem core, and mainly consists on several rdkit functionalities that
characterize small molecules which would later be used for docking. They are inspired on some of the
<a class="reference external" href="https://projects.volkamerlab.org/teachopencadd/">TechOpenCADD talktorials</a></p>
<p>We included filters based on ADME or PAINS qualities; based on shape or fingerprint distance against a target ligand;
and finally tools for generating, managing and filtering by pharmacophores.</p>
</section>
<section id="docking">
<h2>5) Docking<a class="headerlink" href="#docking" title="Link to this heading"></a></h2>
<section id="autodock-index">
<h3>5.1) <a class="reference internal" href="plugins/autodock/index.html#autodock-index"><span class="std std-ref">scipion-chem-autodock</span></a><a class="headerlink" href="#autodock-index" title="Link to this heading"></a></h3>
<p>One of the main plugins in the Scipion-chem framework is the one for AutoDock. Here, we include the main docking
functionalities of AutoDock4 and Vina, together with AutoLigand for protein pocket prediction and two protocols for
the preparation of the receptor and the ligands (<a class="reference external" href="https://github.com/forlilab/Meeko">Meeko</a>).
At the moment of this writing, we have just included the
<a class="reference external" href="https://github.com/ccsb-scripps/AutoDock-GPU">AutoDockGPU</a> and the
<a class="reference external" href="https://github.com/ccsb-scripps/AutoDock-Vina">Vina-1.2</a> versions from the ForliLab (<a class="reference external" href="https://forlilab.org/">https://forlilab.org/</a>)</p>
<p>Scipion-chem-autodock automatically manages the necessary pdbqt format files and interconverts them from and to other
file formats when necessary using openbabel or AutoDockTools when necessary. This way, the interoperability with other
VDS tools can be achieved.</p>
</section>
<section id="scipion-chem-lephar">
<h3>5.2) <strong>scipion-chem-lephar</strong><a class="headerlink" href="#scipion-chem-lephar" title="Link to this heading"></a></h3>
<p>With this plugin from <a class="reference external" href="http://www.lephar.com/software.htm">Lephar</a>, we include yet another tool for docking, LeDock.
Together with the main docking tool, we include another protocol for target preparation using LePro.</p>
</section>
<section id="scipion-chem-rosetta">
<h3>5.3) <strong>scipion-chem-rosetta</strong><a class="headerlink" href="#scipion-chem-rosetta" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://www.rosettacommons.org/">Rosetta</a> is widely known as one of the main software for molecular modelling.
Even though this offers a lot of useful functionalities, as for today only protocols related to DARC docking are
integrated. This includes a protocol for the target preparation and the protocol for <a class="reference external" href="https://www.rosettacommons.org/docs/latest/application_documentation/docking/DARC">DARC docking</a>
Regarding EM and protein modelling, this plugin also includes a protocol to fit a protein reference structure into an
electron density map.</p>
<p>As for the rest of the plugins, the Scipion team is open to hear any suggestion or accept any help for integrating
new functionalities, so let us know if you have any.</p>
<p>Also, user must know that the installation of Rosetta is not automatic. Instruction for a proper installation can be
found in the <a class="reference external" href="https://github.com/scipion-chem/scipion-chem-rosetta">scipion-chem-rosetta</a> repo.</p>
</section>
</section>
<section id="molecular-dynamics">
<h2>6) <strong>Molecular Dynamics</strong><a class="headerlink" href="#molecular-dynamics" title="Link to this heading"></a></h2>
<p>Molecular Dynamics (MD) simulate the state and movement of some molecules through time by modelling the energy
interactions of their atoms and calculating the coordinates and velocities of each of them during a fixed number of
steps. This modelling is never straight forward, and there are many parameters that must be tuned, but if it’s done
properly one might get an accurate enough representation of the molecule in the real environment.</p>
<p>In Scipion-chem, we have included several programs that perform MD to model and simulate atomic
structures (usually proteins) and even protein-ligand complexes.</p>
<p>In the VDS context, once you have some promising molecules docked to your receptor, one might want to do further
computational studies to check whether the predicted interaction will be maintained in time or not.
This can be achieved using molecular dynamics on the receptor-ligand complex.</p>
<section id="scipion-chem-amber">
<h3>6.1) <strong>scipion-chem-amber</strong><a class="headerlink" href="#scipion-chem-amber" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://ambermd.org/">Amber</a> is a suite of biomolecular simulation programs. It contains open software for preparing
and running MD simulations (AmberTools) and a second part which requires a license (Amber).
By the time this documentation is writen, only AmberTools is integrated in Scipion-chem, so it can be installed
automatically.</p>
<p>AmberTools also include cpptraj, a tool for managing and analyzing the resulting MD trajectories and is also
being integrated in Scipion-chem-amber.</p>
</section>
<section id="scipion-chem-gromacs">
<h3>6.2) <strong>scipion-chem-gromacs</strong><a class="headerlink" href="#scipion-chem-gromacs" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://www.gromacs.org/">Gromacs</a> A free and open-source software suite for high-performance MD and
output analysis.
By the time this documentation is writen, only the functionality for protein simulations is integrated. The
parametrization of small molecules for simulating complexes will be handled and incorporated soon.</p>
<p>Different Gromacs tools are also included to manage and analyze the resulting trajectories.</p>
</section>
<section id="scipion-chem-pwchemschrodinger">
<h3>6.3) <strong>scipion-chem-pwchemSchrodinger</strong><a class="headerlink" href="#scipion-chem-pwchemschrodinger" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://www.schrodinger.com/">Schrodinger</a> is a full platform itself devoted to drug discovery and materials design.
From Scipion-chem, we have integrated some of its tools and adapted them to or workflow, managing their maestro format
files. Among the utils integrated, we have receptor (prepwizard) and ligand preparation (LigPrep),
binding pocket prediction (SiteMap), docking (Glide) and molecular dynamics (Desmond).</p>
<p>This way, the user can design a full workflow of VDS using Schrodinger from the Scipion framework and analyze its
results from the incorporated viewers.</p>
<p>Finally, user must know that the installation of Schrodinger is not automatic, since it depends on a license.
Instruction for a proper installation can be found in the
<a class="reference external" href="https://github.com/scipion-chem/scipion-chem-pwchemSchrodinger">scipion-chem-pwchemSchrodinger</a> repo.</p>
</section>
</section>
<section id="others">
<h2>7) <strong>Others</strong><a class="headerlink" href="#others" title="Link to this heading"></a></h2>
<p>Further functionalities outside the most common VDS workflow are included in Scipion-chem. They include:</p>
<section id="scipion-chem-modeller">
<h3>7.1) <strong>scipion-chem-modeller</strong><a class="headerlink" href="#scipion-chem-modeller" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://salilab.org/modeller/">Modeller</a> is a tool used for protein homology modelling. In this plugin,
we include a protocol for performing the typical <a class="reference external" href="https://salilab.org/modeller/manual/node15.html">homology modelling</a>
based on the sequence alignment of reference structures; and a protocol for <a class="reference external" href="https://salilab.org/modeller/wiki/Mutate_model">modelling the structural change of
mutations</a> in the protein sequence.</p>
</section>
<section id="scipion-chem-blast">
<h3>7.2) <strong>scipion-chem-blast</strong><a class="headerlink" href="#scipion-chem-blast" title="Link to this heading"></a></h3>
<p><a class="reference external" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">Blast</a> is the NCBI widely known program for searching biological
sequences.
In this plugin, we include tools for managing and configuring blast searchs, creating local BLAST databases and
easily downloading NCBI data (protein/nucleotide sequences or small molecules) from their IDs</p>
<blockquote>
<div><div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
</div>
</div></blockquote>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="plugins/pwchem/index.html" class="btn btn-neutral float-right" title="scipion-chem" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Scipion Chem team.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>